lib/pymedphys/_experimental/streamlit/apps/transfer_check.py

```diff
@@ -70,185 +58,405 @@ def main():
-        remove = []
-        for i in mosaiq_table.iloc[:].index:
-            if i not in index:
-                remove.append(i)
-
-        # Drop all indices in the remove list to get rid of fields irrelevant for this comparison
-        mosaiq_table = mosaiq_table.drop(remove)
-        mosaiq_table = mosaiq_table.sort_index(axis=1)
-        mosaiq_table = mosaiq_table.sort_values(by=["field_label"])
-
-        # Compare values between the two systems and create a new dataframe with the results
-        results = compare_to_mosaiq(dicom_table, mosaiq_table)
-        results = results.transpose()
+def show_field_rx(dicom_table, selected_label):
+    st.write(
+        "**RX**: ",
+        dicom_table[dicom_table["field_label"] == selected_label.values[0]][
+            "rx"
+        ].values[0],
+    )
 
-        # Create a radio selection of prescriptions in mosaiq to choose from for displaying results
-        rx_selection = st.radio("Select RX: ", mosaiq_table.site.unique())
-        rx_fields = mosaiq_table[mosaiq_table["site"] == rx_selection][
-            "field_name"
-        ].values
 
-        # create a radio selection of fields to compare, only fields within selected rx appear as choices
-        field_selection = st.radio("Select field to compare:", rx_fields)
-        selected_label = mosaiq_table[mosaiq_table["field_name"] == field_selection][
-            "field_label"
-        ]
-        dicom_field_selection = dicom_table[
-            dicom_table["field_label"] == selected_label.values[0]
-        ]["field_name"].values[0]
-        st.subheader("Comparison")
+def show_comparison_of_selected_fields(dicom_field_selection, results):
+    dicom_field = str(dicom_field_selection) + "_DICOM"
+    mosaiq_field = str(dicom_field_selection) + "_MOSAIQ"
+    display_results = results[[dicom_field, mosaiq_field]]
 
-        # If a field is selected, write a side by side comparison of the DICOM and MOSAIQ plan information
-        if len(field_selection) != 0:
+    display_results = display_results.drop(
+        ["dob", "first_name", "last_name", "mrn"], axis=0
+    )
 
-            # Write prescription listed in the RP file
-            st.write(
-                "**RX**: ",
-                dicom_table[dicom_table["field_label"] == selected_label.values[0]][
-                    "rx"
-                ].values[0],
-            )
+    display_results = display_results.style.apply(colour_results, axis=1)
+    st.dataframe(display_results.set_precision(2), height=1000)
+
+
+def get_structure_aliases():
+    cwd = os.getcwd().replace("\\", "/")
+    file_path = cwd + "/lib/pymedphys/_experimental/chartchecks/ALIASES.csv"
+    alias_df = pd.read_csv(file_path)
+    for i in range(len(alias_df.keys())):
+        df_list = alias_df.iloc[0][i][1:-1].split((","))
+        formatted_df_list = []
+        for item in df_list:
+            formatted_df_list.append(item.replace("'", "").strip(" "))
+        alias_df.iloc[0][i] = formatted_df_list
```

Comment:
Regarding the pylint message, instead of using enumerate as it suggested, I would recommend to swap to using iteritems. See:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iteritems.html

Then instead of calling `alias_df.iloc[0][i]` utilise the column content returned by that above method.

Also... as a side note, I would recommend no replacing the dataframe inplace (`alias_df.iloc[0][i] = formatted_df_list`) instead, create a new dataframe that has the formatting be undergone. This is more a decision of taste, but I find in-place changing of dataframes can get one in trouble down the line.

I still do sometimes do in-place replace dataframes. But it can make it quite hard to reason about what is going on. An example of what can happen... you could accidently be changing the output of a cached streamlit function for example.

Reply:
I've gone ahead and switched the Aliases to json instead of csv, and this actually simplified things to where I no longer have this problem! I also fixed the other enumerate issues.

Any recommendations on the too many nested-blocks issue?

Reply:
> Any recommendations on the too many nested-blocks issue?

I'm just on my phone, so can't see the actual code itself easily. But if you've got too much going on in a function I'd recommend spinning out some sections of the code into separate functions.